#! /bin/sh -e

# DP: test_sundry: Don't fail on import of the profile and pstats module

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Lib/test/test_sundry.py~	2008-05-22 18:02:34.000000000 +0200
+++ Lib/test/test_sundry.py	2008-05-22 18:03:58.000000000 +0200
@@ -65,7 +65,11 @@
             import os2emxpath
             import pdb
             import posixfile
-            import pstats
+            try:
+                import pstats  # separated out into the python-profiler package
+            except ImportError:
+                if test_support.verbose:
+                    print "skipping profile and pstats"
             import py_compile
             import rexec
             import rlcompleter
